### ISSUE TITLE
how to save the random survival forest model without data?

### ISSUE THREAD

**Issue by willthd** [kogalur/randomForestSRC#52]
I'm using randomForestSRC.
But when export the model, the model object has data...
How can i save the only model without data?

**Comment by kogalur** [comment]
Well, this is an interesting question.  The reason we include the training data is to make restoration of the model easier during prediction.  If you never intend to use it for prediction, I suppose we could save the model without the training data.  In addition, in big data situations, it might actually be preferable not to include the training data.  But if training data is not included in the model and one does want to conduct prediction, one would have to save terminal node statistics (TNS).  This is fairly simple in regression, and a little more complicated in classification, but in survival, the  data we need to save becomes quite large.  We have a hidden parameter "terminal.quants" in the rfsrc() call that saves TNS. Its default value is FALSE.  I suppose it's possible for you to turn this on and then delete the training data post export.  But it's not clear to me that you can do prediction on the resulting model, as we might be extracting other necessary information from the training data during prediction.

**Comment by JMason823** [comment]
@kogalur I am also interested in saving a model built with randomForestSRC without the train data and then later using it for prediction purposes. The reasoning is because I have successfully built models on an environment in which the individual subject data cannot leave. I have tried building a model with "terminal.quants = TRUE", manually deleting the instances of "xvar" within the model, and then predicting (which fails). Can you offer guidance on this?

**Comment by kogalur** [comment]
I think it depends on how you manually deleted the data.  What kind of error are you experiencing?  It might be that the system expects data during some preliminary checks even if it doesn't use it.  The way around that would be to send in a dummy array, though I'm not sure even that would work.  I think the capability you ask for is very reasonable, and we'll put it on our do-to list for a future release.  Unfortunately, we can't give you an estimate of where this campaign would reside on our priority list at this time.

**Comment by JMason823** [comment]
I am able to successfully predict on my test set via:
     prediction <- predict(model, test_set)

I manually delete the data via:
     model$xvar <- NULL
     model$forest$xvar <- NULL

When I run the same prediction line from above, I get the following error.
     Error in rep("R", ncol(dat)) : invalid 'times' argument

Looking forward to the future release that supports this feature. And if it matters, this is in reference to survival models.

**Comment by kogalur** [comment]
We have reopened this issue, after prematurely closing it.  We are looking into implementing a minimal forest object and will update this thread as we make progress.  The goal is to allow users to save, restore and predict using a model that does not contain the embedded training data.

**Comment by bnuzyc91** [comment]
@kogalur Thank you so much for working on this issue. it is a great feature! We are preparing a manuscript about the survival analysis using randomForestSRC and hope we can share our final model (tree structure/risk prediction based on the forest) without sharing the training data. Could you please give me a rough estimate of when this feature will be available? Thanks again!

**Comment by kogalur** [comment]
@bnuzyc91 We estimate six weeks, but we'll give you a beta build on which you can test your scenario, before that.

**Comment by kogalur** [comment]
Here is the source package for a beta build, RF-SRC 2.10.1.5 containing rfsrc.anonymous().  You will need the appropriate toolchain in place to be able to compile and install R packages on your platform.  Please see the Rd help file on how to use this new function.  Thanks.  

https://www.dropbox.com/s/oz3tq99q685yfta/randomForestSRC_2.10.1.5.tar.gz?dl=0
